### ISSUE TITLE
Question about centroids update

### ISSUE THREAD

**Issue by xyy19920105** [zhmiao/OpenLongTailRecognition-OLTR#4]
Thank you for releasing the code for this awesome work.
I have a question about the centroids update. I have read the code and find that centroids only calculated once at the beginning of the model initialization at stage 2. Could you help me to find out how the centroids update? And I wonder that is the centroids correct because the parameters of attention is just initialized and the features of centroids are not learned.

**Comment by zhmiao** [comment]
Yes. Thank you very asking. Please refer to this issue: https://github.com/zhmiao/OpenLongTailRecognition-OLTR/issues/3#issue-444830256 . It is about the similar questions you are asking I think.

**Comment by xyy19920105** [comment]
The centroids is initialized by the features of train data, and updated by the training stage directly. Is that right? Do centroids have no relations with the features of train data after initializing?

**Comment by zhmiao** [comment]
@xyy19920105 In the second training stage, the original training data is still used for training, and the centroids are updated according to the modified center loss.

**Comment by EmmaW8** [comment]
Emmm...
Is my understanding right?

The authors first trained a plain model then utilized this trained model to calculate the initialization of the centroids.
Therefore, we need to load weights of the plain model to the final model with MetaEmbedding then train the final model based on these weights. Since the centroids are the parameters of the MetaEmbedding model, they can be updated automatically with the loss back-propagation.

Just wondering is it a fair contrast between plain model and MetaEmbedding model?

**Comment by liuziwei7** [comment]
From the perspective of machine learning or meta learning, the comparison is fair as long as these models have undergone the same number of (sufficient) gradient updates, and with the same amount of training data or episode experience. 

The plain model has already converged in the first stage training, further iterations will not improve or even hamper its performance.

Hope this could help :)

**Comment by zhmiao** [comment]
Since it has been more than 20 days. I will close this issue for now. If you are having more questions, we will open it again. Thank you very much.
